src/api/data/helm/client/client.go

```diff
@@ -0,0 +1,140 @@
+package client
+
+import (
+	"fmt"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/helm/monocular/src/api/config"
+	"github.com/helm/monocular/src/api/data"
+	releasesapi "github.com/helm/monocular/src/api/swagger/restapi/operations/releases"
+	"k8s.io/helm/cmd/helm/installer"
+	"k8s.io/helm/pkg/helm"
+	"k8s.io/helm/pkg/helm/portforwarder"
+	"k8s.io/helm/pkg/kube"
+	rls "k8s.io/helm/pkg/proto/hapi/services"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
+	"k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset/typed/core/internalversion"
+	"k8s.io/kubernetes/pkg/labels"
+
+	helmreleases "github.com/helm/monocular/src/api/data/helm/releases"
+	kerrors "k8s.io/kubernetes/pkg/api/errors"
+	"k8s.io/kubernetes/pkg/client/restclient"
+)
+
+const (
+	tillerNamespace   = "kube-system"
+	tillerServiceName = "tiller-deploy"
+	tillerPort        = 44134
+)
+
+type helmClient struct {
+	portForward bool
+}
+
+// NewHelmClient returns the Helm implementation of data.Client
+func NewHelmClient() data.Client {
+	conf, _ := config.GetConfig()
+	return &helmClient{
+		portForward: conf.TillerPortForward,
+	}
+}
+
+func (c *helmClient) ensureTillerInstalled(namespace string, client *internalclientset.Clientset) error {
+	_, err := getFirstRunningPod(client, namespace, labels.Set{"app": "helm", "name": "tiller"}.AsSelector())
+	if err == nil {
+		return nil
+	}
+
+	log.WithFields(log.Fields{"error": err.Error()}).Info("Can't connect to Tiller. Installing")
+	if err = installer.Install(client, namespace, "", false, false); err != nil {
+		if !kerrors.IsAlreadyExists(err) {
+			return fmt.Errorf("error installing Tiller: %s", err)
+		}
+	}
+	return fmt.Errorf("Can't connect to Tiller. Installing, please wait")
+}
+
+// InitializeClient returns a helm.client
+func (c *helmClient) initialize() (*helm.Client, error) {
+	// From helm.setupConnection
+	config, client, err := getKubeClient("")
+	if err != nil {
+		return nil, err
+	}
+
+	err = c.ensureTillerInstalled(tillerNamespace, client)
+	if err != nil {
+		return nil, err
+	}
+
+	var tillerHost = fmt.Sprintf("%s.%s:%d", tillerServiceName, tillerNamespace, tillerPort)
+
+	if c.portForward {
+		tunnel, err := portforwarder.New(tillerNamespace, client, config)
```

Comment:
If the developer enabled portForwarding, a tunnel will be created in every request. Default is to use the new tiller-deploy service.